src/main/java/at/hannibal2/skyhanni/api/EffectApi.kt

```diff
@@ -0,0 +1,314 @@
+     */
+    private val potionRemainingLoreTimerPattern by RepoPattern.pattern(
+        "inventory.effects.effecttimeleft",
+        "§7Remaining: §f(?<time>[\\d:]+)",
+    )
+
+    /**
+     * REGEX-TEST:  Repellent: §r§9MAX §r§7(12s)
+     */
+    private val repellentPattern by RepoPattern.pattern(
+        "misc.nongodpot.repellant",
+        " Repellent: §r§[97a](?<tier>\\w+)?(?: §r§7\\((?<time>\\d)s\\))?",
+    )
+    // </editor-fold>
+
+    private val profileStorage get() = ProfileStorageData.profileSpecific
+
+    // TODO move the whole list into the repo
+    enum class NonGodPotEffect(
+        val tabListName: String,
+        val isMixin: Boolean = false,
+        val inventoryItemName: String = tabListName,
+    ) {
+        SMOLDERING("§aSmoldering Polarization I"),
+        GLOWY("§2Mushed Glowy Tonic I"),
+        WISP("§bWisp's Ice-Flavored Water I"),
+        GOBLIN("§2King's Scent I"),
+
+        INVISIBILITY("§8Invisibility I"), // when wearing sorrow armor
+
+        REV("§cZombie Brain Mixin", true),
+        TARA("§6Spider Egg Mixin", true),
+        SVEN("§bWolf Fur Mixin", true),
+        VOID("§6End Portal Fumes", true),
+        BLAZE("§fGabagoey", true),
+        GLOWING_MUSH("§2Glowing Mush Mixin", true),
+        HOT_CHOCOLATE("§6Hot Chocolate Mixin", true),
+
+        DEEP_TERROR("§4Deepterror", true),
+
+        GREAT_SPOOK("§fGreat Spook I", inventoryItemName = "§fGreat Spook Potion"),
+
+        DOUCE_PLUIE_DE_STINKY_CHEESE("§eDouce Pluie de Stinky Cheese I"),
+
+        HARVEST_HARBINGER("§6Harvest Harbinger V"),
+
+        PEST_REPELLENT("§6Pest Repellent I§r"),
+        PEST_REPELLENT_MAX("§6Pest Repellent II"),
+
+        CURSE_OF_GREED("§4Curse of Greed I"),
+
+        COLD_RESISTANCE_4("§bCold Resistance IV"),
+
+        POWDER_PUMPKIN("§fPowder Pumpkin I"),
+        FILET_O_FORTUNE("§fFilet O' Fortune I"),
+        CHILLED_PRISTINE_POTATO("§fChilled Pristine Potato I"),
+    }
+
+    // Todo : cleanup and add support for poison candy I, and add support for splash / other formats
+    @HandleEvent(onlyOnSkyblock = true)
+    @Suppress("MaxLineLength")
+    fun onChat(event: SkyHanniChatEvent) {
+        hotChocolateMixinConsumePattern.matchMatcher(event.message) {
+            val durationAdded = TimeUtils.getDuration(group("time"))
+            EffectDurationChangeEvent(
+                NonGodPotEffect.HOT_CHOCOLATE,
+                EffectDurationChangeType.ADD,
+                durationAdded,
+            ).post()
+        }
+        godPotConsumePattern.matchMatcher(event.message) {
+            val durationAdded = TimeUtils.getDuration(group("time"))
+            val existingValue = profileStorage?.godPotExpiry?.takeIfInitialized() ?: SimpleTimeMark.now()
+            profileStorage?.godPotExpiry = existingValue + durationAdded
+        }
+
+        val effect: NonGodPotEffect?
+        val changeType: EffectDurationChangeType?
+        val duration: Duration?
+
+        when (event.message) {
```

Comment:
there is only one `remove`, everything else is `set`. maybe we can split the `remove` out of the `when`, then return a pair of `effecttype` and `duration` in the `when`.

Reply:
i just saw the todo comment, maybe this is for another pr